### PR TITLE
Add Japanese language

### PR DESCRIPTION
**Pull request by maboroshin**
"Files changed" had five files. Stopped "git add -A". Pull requests for Japanese language files only.

### CODE REVIEW

**Comment by staniel359** [comment]
@maboroshin So I guess you want to see your visualized translations.
But I can't release a version with incomplete translation.
You may want to setup and run muffon's local version?
I can give you the instructions.

File: src/plugins/i18n/locales/ja.json
Comment (staniel359) [inline src/plugins/i18n/locales/ja.json]: @maboroshin Please move Electron translations to `muffon/electron/plugins/i18n/locales` folder.

**Review by staniel359 — CHANGES_REQUESTED** [review]

File: src/plugins/i18n/locales/ja.json
Comment (maboroshin) [reply src/plugins/i18n/locales/ja.json:2]: 72% (145/510) : There are 510 texts, untranslated is 145.

**Comment by maboroshin** [comment]
72% (145/510) : There are 510 texts, untranslated is 145.

**Comment by staniel359** [comment]
@maboroshin Okay, take your time.

**Comment by maboroshin** [comment]
Other than mffon, I know of no other project that requires 100%.

**Comment by staniel359** [comment]
@maboroshin What do you mean?
Are there some English words that can't be translated to Japanese?

**Comment by maboroshin** [comment]
The most stringent TRANSMISSION was based on 95%.

There is no English word that cannot be translated.
I am checking the actual display.  So mistranslations are unlikely to occur. This is why it takes time. It takes extra time to understand the untranslated parts.

File: src/plugins/i18n/locales/ja.json
Comment (maboroshin) [reply src/plugins/i18n/locales/ja.json]: @staniel359 I moved.

Comment (staniel359) [reply src/plugins/i18n/locales/ja.json]: @maboroshin Still there.

**Comment by maboroshin** [comment]
Sites using outside translation tools are often released with incomplete translations.
I sometimes don't make it 100% to avoid mistranslation.

**Comment by staniel359** [comment]
@maboroshin There should be no mistranslations as translations are grouped logically.

And this app uses 100% of the translations.

**Comment by maboroshin** [comment]
Sites using outside translation tools are often released with incomplete translations.
I sometimes don't make it 100% to avoid mistranslation.

My Japanese translation of yattee is 97%. This is because I do not use this app. I do not translate without actually using it. Forcing it to 100% would be prone to mistranslation.  e.g. "Clear" would be "weather clear" ( exaggerated example).
https://hosted.weblate.org/projects/yattee/localizable-strings/ja/

Check the translation status of Auto Dark mode. 100% is usually not maintained.
https://hosted.weblate.org/projects/yattee/localizable-strings/
Or ksnip (The development period is long and there are many translated strings.)
https://hosted.weblate.org/projects/ksnip/#languages

File: src/plugins/i18n/locales/ja.json
Comment (maboroshin) [reply src/plugins/i18n/locales/ja.json]: It exists here in my browser.
image

**Comment by maboroshin** [comment]
Understood. I will abort.

**Comment by staniel359** [comment]
@maboroshin Why?

**Comment by maboroshin** [comment]
@staniel359 It takes time to find untranslated parts. This app has too much time than other software.
With [invidious](https://yewtu.be/) and [Piped](https://piped.kavin.rocks/), all I have to do is translate the untranslated parts again, because a week later they show up in the actual UI.
I am checking to make sure I am not mistranslating. I have to use the app or read the source code to see where it appears.

Even with lossless cut I do not translate 100%. Once the translations are merged and the next version is released, I will translate the rest. lossless cut does not require 100%, does it?
https://hosted.weblate.org/projects/losslesscut/losslesscut/

**Comment by staniel359** [comment]
@maboroshin Okay, then please translate as much as possible.

And if you want to see your translations in action, I can create a test release with your translations for you.

**Comment by maboroshin** [comment]
I'm in the middle of translating, but I saw the 1.5.0Update, so I did a pull request.

The only software that requires percentages is TRANSMISSION. If it is less than 95%, the translation is invisible. There is very little software of this kind.

https://github.com/crow-translate/crow-translate
CROW translations can easily be reflected in the UI. No need to wait for release. Not 100% as I don't know how "Print" actually works. It does not seem to be printer related.

**Comment by staniel359** [comment]
@maboroshin So are you going to continue translating?

**Comment by maboroshin** [comment]
Umm... When it takes time, I move on to the next project once.

So far I have discontinued muffon and have completed 3 projects.
https://github.com/Moosync/Moosync
https://github.com/stephane-r/holoplay-pwa
https://github.com/amd64fox/SpotX

**Comment by staniel359** [comment]
@maboroshin Okay.

**Comment by staniel359** [comment]
@maboroshin Hello.
Maybe you would like to continue your translation work?
I can provide you a live version.

**Comment by maboroshin** [comment]
Hi! Is this the version that reflects the Japanese translation? If there is a Live version, I would like to do it.

**Comment by staniel359** [comment]
@maboroshin I haven't added Japanese language yet, but you can add and test it locally.
I can give you the instructions.

**Comment by maboroshin** [comment]
OKey. if it is easy I will do it.

**Comment by staniel359** [comment]
@maboroshin
1. Install Node.js and Yarn
2. `git clone https://github.com/staniel359/muffon.git`
3. `cd muffon/`
4. Create `.env` file with `VITE_APP_ANONYMOUS_TOKEN=<some_random_token>` (then send me this token on my email)
5. In `src/plugins/axios.js` on line 3 change `const isDevelopment` to `false`
6. Add your translation files (`jp.json`) in `src/plugins/i18n/locales/` and `electron/plugins/i18n/locales/`
7. Search for `// i18n` in project folder and make changes in these files to add Japanese language.
8. `yarn`
9. `yarn start` (in one window)
10. `yarn start:electron` (in second window)

**Comment by maboroshin** [comment]
Thank you. This information would be fine in the README.

I put off projects that take a long time. I will consider doing this after other time consuming projects.

**Comment by staniel359** [comment]
@maboroshin Okay.

**Comment by staniel359** [comment]
@maboroshin Hey, how's it going?
Would you like to continue your work maybe?

**Comment by xyloflake** [comment]
Hey @maboroshin, I've added the Japanese language translation now, would like to test and rate it?

**Comment by maboroshin** [comment]
Thank you. It was this one.
https://github.com/staniel359/muffon/blob/main/src/plugins/i18n/locales/ja.json

**Comment by xyloflake** [comment]
@maboroshin I mean can you use the app and provide feedback for the translation? I have Japanese N1 qualification, but I'm not native so your feedback would be useful. You can change the language in settings.

**Comment by maboroshin** [comment]
@xyloflake : I'm surprised that you are not native. You are better than a beginner translator. 
N1, This is the highest level of the Japanese-Language Proficiency Test!

---

Too many transliterations. So I apply more common Japanese. This will take time.

**Comment by xyloflake** [comment]
@maboroshin はは、褒めてくれてありがとう。日本語は２ヶ月間勉強して、準備ができたと感じたら試験を受けたんだ。漢字は約４００個しか覚えられなかったけど、それで十分だったみたいで、今はN1の資格を持っているよ。それと、アプリを使ったことある？もし使ったら、もしエラーがあれば報告してくれると嬉しいな。エラーがなければ教えてくれて、今は開発者の一員として参加しているんだ（笑）。


Also sorry for using informal japanese if that offends you, I use that more often than formal one, so I am better at it 😅

**Comment by xyloflake** [comment]
@maboroshin have you aborted by the way? Because I've translated everything.

**Comment by maboroshin** [comment]
> have you aborted by the way? 

Staniel asked for 100% translation rate. So I aborted. So I have given this a low priority. But I'll take this opportunity to review it.

**Comment by xyloflake** [comment]
@maboroshin thanks, do you have discord?

**Pull request closed without merging**
